engine/common/follower/pending_tree/pending_tree.go

```diff
@@ -147,28 +141,38 @@ func (t *PendingTree) AddBlocks(certifiedBlocks []CertifiedBlock) ([]CertifiedBl
 func (t *PendingTree) connectsToFinalizedBlock(block CertifiedBlock) bool {
 	if block.Block.Header.ParentID == t.lastFinalizedID {
 		return true
-	} else if parentVertex, found := t.forest.GetVertex(block.Block.Header.ParentID); found {
+	}
+	if parentVertex, found := t.forest.GetVertex(block.Block.Header.ParentID); found {
 		return parentVertex.(*PendingBlockVertex).connectedToFinalized
-	} else {
-		return false
 	}
+	return false
 }
 
 // FinalizeForkAtLevel takes last finalized block and prunes levels below the finalized view.
 // When a block is finalized we don't care for all blocks below it since they were already finalized.
+// Finalizing a block might result in observing a connected chain of blocks that previously weren't.
+// These blocks will be returned as result of invocation.
```

Comment:
That is very subtle. Would suggest to add the following detailed explanation:

```suggestion
// Finalizing a block might causes the pending PendingTree to detect _additional_ blocks as now
// being connected to the latest finalized block. This happens of some connecting blocks are missing
// and then a block higher than the missing blocks is finalized.
// In the following example, B is the last finalized block known to the PendingTree
//
//	A ← B  ←-?-?-?-- X ← Y ← Z
//
// The network has already progressed to finalizing block X. However, the interim blocks denoted
// by '←-?-?-?--' have not been received by our PendingTree. Therefore, we still consider X,Y,Z
// as disconnected. If the PendingTree tree is now informed that X is finalized, it can fast-
// forward to the respective state, as it anyway would prune all the blocks below X.
//
// If the PendingTree detect additional blocks as descending from the latest finalized block, it
// returns these blocks. Returned blocks are ordered such that parents appear before their children.
//
```